src/Microsoft.DotNet.Darc/src/DarcLib/LocalGitClient.cs

```diff
@@ -149,46 +149,60 @@ public Task MergePullRequestAsync(string pullRequestUrl, MergePullRequestParamet
         /// <returns></returns>
         public async Task CommitFilesAsync(List<GitFile> filesToCommit, string repoUri, string branch, string commitMessage)
         {
-            foreach (GitFile file in filesToCommit.Where(f => f.Operation != GitFileOperation.Delete))
+            string repoDir = LocalHelpers.GetRootDir(_logger);
+            try
             {
-                switch (file.Operation)
+                using (LibGit2Sharp.Repository localRepo = new LibGit2Sharp.Repository(repoDir))
                 {
-                    case GitFileOperation.Add:
-                        string parentDirectory = Directory.GetParent(file.FilePath).FullName;
-
-                        if (!Directory.Exists(parentDirectory))
+                    foreach (GitFile file in filesToCommit.Where(f => f.Operation != GitFileOperation.Delete))
+                    {
+                        switch (file.Operation)
                         {
-                            Directory.CreateDirectory(parentDirectory);
-                        }
+                            case GitFileOperation.Add:
+                                string parentDirectory = Directory.GetParent(file.FilePath).FullName;
 
-                        string fullPath = Path.Combine(repoUri, file.FilePath);
-                        using (var streamWriter = new StreamWriter(fullPath))
-                        {
-                            string finalContent;
-                            switch (file.ContentEncoding)
-                            {
-                                case ContentEncoding.Utf8:
-                                    finalContent = file.Content;
-                                    break;
-                                case ContentEncoding.Base64:
-                                    byte[] bytes = Convert.FromBase64String(file.Content);
-                                    finalContent = Encoding.UTF8.GetString(bytes);
-                                    break;
-                                default:
-                                    throw new DarcException($"Unknown file content encoding {file.ContentEncoding}");
-                            }
-                            finalContent = NormalizeLineEndings(fullPath, finalContent);
-                            await streamWriter.WriteAsync(finalContent);
-                        }
-                        break;
-                    case GitFileOperation.Delete:
-                        if (File.Exists(file.FilePath))
-                        {
-                            File.Delete(file.FilePath);
+                                if (!Directory.Exists(parentDirectory))
+                                {
+                                    Directory.CreateDirectory(parentDirectory);
```

Comment:
This method already check if the directory exist. No need for the check on line 164.

Reply:
I'm only changing indentation in this section of the file, and I think it actually makes the code a little more readable with the check. Let me know if you feel strongly and I'll revisit.